src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraphUtils.cs

```diff
@@ -85,5 +87,32 @@ private static List<ExecutionPlanGraphPropertyBase> GetProperties(PropertyDescri
             }
             return propsList;
         }
+
+        private static List<ExecutionPlanRecommendation> ParseRecommendations(ShowPlanGraph.ShowPlanGraph g, string fileName)
+        {
+            if (g.Description != null && g.Description.MissingIndices != null)
+            {
+                return g.Description.MissingIndices.Select(mi => new ExecutionPlanRecommendation
+                {
+                    DisplayString = mi.MissingIndexCaption,
+                    QueryText = mi.MissingIndexQueryText,
+                    FormattedQueryText = ParseMissingIndexQueryText(fileName, mi.MissingIndexImpact, mi.MissingIndexDatabase, mi.MissingIndexQueryText)
+                }).ToList();
+            }
+            return null;
+        }
+
+        private static string ParseMissingIndexQueryText(string doctitle, string impact, string database, string query)
```

Comment:
XML doc for these. 

And maybe update the `doctitle` param - in all the previous methods you call it `fileName` so that seems reasonable to use as the name here too.